File: jscomp/test/installation_test.ml
let suites :  Mt.pair_suites ref  = ref []
let test_id = ref 0
let eq loc x y = 
  incr test_id ; 
  suites := 
    (loc ^" id " ^ (string_of_int !test_id), (fun _ -> Mt.Eq(x,y))) :: !suites




let () = 
  match [%node __dirname] with | Some p -> 
      let root = App_root_finder.find_package_json p in
      let bsc_exe = 
        Node.Path.join 
          [| root ; "lib";"bsc.exe" |] in 

      begin match Node.Child_process.execSync 
              (bsc_exe ^ " -where ") 
              (Node.Child_process.option  ~encoding:"utf8" ()) with 
      | output -> 
        let dir = Js.String.trim output in 
        let files = Node.Fs.readdirSync dir  in
        let exists = 
          files 
          |> Js.Array.indexOf "pervasives.cmi" in 
        let non_exists = 
          files 
          |> Js.Array.indexOf "pervasive.cmi" in 
        let v = (exists >= 0 && non_exists < 0) in
        Js.log v;
      | exception e -> 
        assert false
      end
      | None  ->  assert false 

let () = 
  Mt.from_pair_suites __FILE__ !suites

File: jscomp/test/installation_test.js
'use strict';

var Mt = require("./mt.js");
var Block = require("../../lib/js/block.js");

var suites = {
  contents: /* [] */0
};

var test_id = {
  contents: 0
};

function eq(loc, x, y) {
  test_id.contents = test_id.contents + 1 | 0;
  suites.contents = /* :: */[
    /* tuple */[
      loc + (" id " + String(test_id.contents)),
      (function (param) {
          return /* Eq */Block.__(0, [
                    x,
                    y
                  ]);
        })
    ],
    suites.contents
  ];
  
}

Mt.from_pair_suites("Installation_test", suites.contents);

exports.suites = suites;
exports.test_id = test_id;
exports.eq = eq;
/*  Not a pure module */
